model/histogram/float_histogram.go

```diff
@@ -982,3 +964,72 @@ func (i *allFloatBucketIterator) Next() bool {
 func (i *allFloatBucketIterator) At() Bucket[float64] {
 	return i.currBucket
 }
+
+func targetIdx(idx, schema, targetSchema int32) int32 {
+	return ((idx - 1) >> (schema - targetSchema)) + 1
+}
+
+func convertSpanToSchema(spans []Span, counts []float64, schema, targetSchema int32) ([]Span, []float64) {
+	targetSpans := make([]Span, 0)
+	targetCounts := make([]float64, 0)
+	// the index of bucket for origin schema
+	var bucketIdx int32 = 0
+	var lastTargetBucketIdx int32 = 0
+	// the index of bucket in a span
+	spanBucketIdx := 0
+
+	for i, span := range spans {
+		// determine the boundary of first bucket in this span
+		if i == 0 {
+			bucketIdx = spans[0].Offset
+		} else {
+			bucketIdx += span.Offset
+		}
+		// iterate over each span
+		for j := 0; j < int(span.Length); j, bucketIdx = j+1, bucketIdx+1 {
+			// calculate the target start index
+			targetBucketIdx := targetIdx(bucketIdx, schema, targetSchema)
+
+			switch {
+			case len(targetSpans) == 0:
+				// this is the first span
+				span := Span{
+					Offset: targetBucketIdx,
+					Length: 1,
+				}
+				targetSpans = append(targetSpans, span)
+				targetCounts = append(targetCounts, counts[spanBucketIdx])
+				lastTargetBucketIdx = targetBucketIdx
+
+			case lastTargetBucketIdx == targetBucketIdx:
+				// the target bucket just locates current last span,
+				// then increase counter of last span
```

Comment:
This reads a bit weird. How about: "The current bucket has to be merged into the same target bucket as the previous bucket."